File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0;
        int r=height.size()-1;
        int res=0;
        while(l<r){
            res=max(res,min(height[l],height[r])*(r-l));
            if(height[r]>height[l]){
                l+=1;
            }
            else{
                r-=1;
            }
        }
        return res;
        
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,h=0,len(height)-1
        res=0
        while l<h:
            res=max(res,(h-l)*min(height[l],height[h]))
            if height[l]<height[h]:
                l+=1
            else:
                h-=1
        return res
        